### ISSUE TITLE
could you provide a simple way for using the library with fragments

### ISSUE THREAD

**Issue by m7mdyahia** [michael-rapp/ChromeLikeTabSwitcher#11]
could you provide a simple way for using the library with fragments to be similar to FragmentStatePagerAdapter for example?

**Comment by michael-rapp** [comment]
Do you already have a working solution? It would be useful as a starting point.

**Comment by m7mdyahia** [comment]
I don't have working solution but just thoughts that it could be implemented in a way similar to FragmentStatePagerAdapter

**Comment by michael-rapp** [comment]
I agree that it would probably be a useful feature for some. However, I personally don't need it and therefore this feature has low priority until at least version 1.0.0 of the library has been released. Nevertheless, if someone is willing to work on this or even has prototypical implementation, I am willing to help and merge it eventually.

**Comment by jackflyer** [comment]
I am trying this. 
For my purpose I added a class that extends AbstractState. But app crashes once TabSwitcherView is opened and I select the tab to show up. Because fragment not found.

I need little help on this.
```
private class  TabState extends AbstractState {

      @Override
      public final void saveInstanceState(@NonNull final Bundle outState) {
            // ====> this method gets called when tabSwitcherButton Clicked
      }

      @Override
      public void restoreInstanceState(@Nullable final Bundle savedInstanceState) {
              // This method does not get called. When tab is shown. 
       }

}
```
Can you help me on this? How can I make state so that restoreInstanceState gets called on tab shown?
